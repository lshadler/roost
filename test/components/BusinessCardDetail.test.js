import React from 'react';
import {render} from '@testing-library/react';

import BusinessCardDetail from '../../components/BusinessCardDetail';

describe('BusinessCardDetail Component', () => {
  let businessInfo;

  beforeEach(() => {
    businessInfo = {
      name: 'Heartwork',
      website: 'https://heartwork.com',
      tags: ['coffee', 'test'],
    };
  });
  test('renders in default collapsed state', () => {
    const {asFragment} = render(<BusinessCardDetail
      businessInfo={businessInfo}
      expanded={false}/>);

    expect(asFragment()).toMatchSnapshot();
  });

  test('handles empty tags gracefully', () => {
    delete businessInfo.tags;

    const {asFragment} = render(<BusinessCardDetail
      businessInfo={businessInfo}
      expanded={false}/>);

    expect(asFragment()).toMatchSnapshot();
  });
});
