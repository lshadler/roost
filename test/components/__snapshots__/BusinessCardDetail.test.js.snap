// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BusinessCardDetail Component handles empty tags gracefully 1`] = `
<DocumentFragment>
  <div
    class="MuiCollapse-container MuiBox-root MuiBox-root-39 MuiCollapse-hidden"
    style="min-height: 0px;"
  >
    <div
      class="MuiCollapse-wrapper"
    >
      <div
        class="MuiCollapse-wrapperInner"
      >
        No details yet...
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`BusinessCardDetail Component renders in default collapsed state 1`] = `
<DocumentFragment>
  <div
    class="MuiCollapse-container MuiBox-root MuiBox-root-2 MuiCollapse-hidden"
    style="min-height: 0px;"
  >
    <div
      class="MuiCollapse-wrapper"
    >
      <div
        class="MuiCollapse-wrapperInner"
      >
        <p
          class="MuiTypography-root MuiTypography-body1"
        >
          Tags: coffee, test
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
